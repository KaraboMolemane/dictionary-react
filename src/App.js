import logo from './logo.svg';
import './App.css';
import { useState, useEffect } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faBook } from '@fortawesome/free-solid-svg-icons';


function App() {
  const [error, setError] = useState(null);
  const [isLoaded, setIsLoaded] = useState(false);
  const [items, setItems] = useState([]);
  const [userWord, setUserWord] = useState('');


  useEffect(() => {
    if(userWord !== ''){
      const apiKey = '0478b4ca-8e6d-4800-b8a1-1e7170815716';
      //const apiURL = 'https://www.dictionaryapi.com/api/v3/references/thesaurus/json/umpire?key='+apiKey;     
      //const apiURL = 'https://www.dictionaryapi.com/api/v3/references/thesaurus/json/'+userWord+'?key='+apiKey;
      const apiURL = 'https://www.dictionaryapi.com/api/v3/references/thesaurus/json/'.concat(userWord,'?key=',apiKey);
      //console.log('apiURL;', apiURL);
      
      fetch(apiURL)
        .then(res => res.json())
        .then(
          (result) => {
            setIsLoaded(true);
            setItems(result);
          },
          (error) => {
            setIsLoaded(true);
            setError(error);
          }
        )
    }

  }, [userWord])

  let text = '';
  if (error) {
    text =  'Error:'+error.message;
  } else if (!isLoaded) {
    text = 'Enter your word above';
  } else {
    console.log('items:', items);
    text = <div>{items[0].shortdef[0]}</div>;
  }

  return (
    <div className="container my-5">
    <div className="p-5 text-center bg-body-tertiary rounded-3">
        <FontAwesomeIcon icon={faBook} size='6x' />
        <br />
        <h1 className="text-body-emphasis">Dictionary</h1>
        <p className="col-lg-8 mx-auto fs-5 text-muted">
          This react app allows you to enter a word and it will show you the definition and the example usage. Go ahead and try it!
        </p>
        <div className="bd-example">
      <form>
        <div className="w-50 p-3" style={{marginLeft: '30%'}} >
          <input type="text" className="form-control d-inline-flex align-items-center" id="userInputWord" aria-describedby="emailHelp" placeholder='Type a word here to get the definition'/>          
        </div>       
      </form>
      <h4>{userWord}</h4>
      <span>{text}</span>
      </div>
        <br />
        <div className="d-inline-flex gap-2 mb-5">
            <button className="d-inline-flex align-items-center btn btn-primary btn-lg px-4 rounded-pill" type="button" onClick={() => setUserWord(document.getElementById('userInputWord').value)}>
                Get definition
            </button>
        </div>
    </div>
</div>
);
}

export default App;
