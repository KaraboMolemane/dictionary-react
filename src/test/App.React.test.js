import React from 'react';
import App from '../App';
import renderer from 'react-test-renderer';

test('renders correctly', () => {
    const tree = renderer
    .create(<App />)
    .toJSON();
    expect(tree).toMatchSnapshot();
    });

    test('the data is defined', () => {
        return fetch('https://www.dictionaryapi.com/api/v3/references/collegiate/json/test/a38b9950-2c3e-446c-94bd-02c7c25cbf22').then(data => {
          expect(data).toBeDefined();
        });
      });