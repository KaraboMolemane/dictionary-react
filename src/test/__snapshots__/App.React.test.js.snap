// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container my-5"
>
  <div
    className="p-5 text-center bg-body-tertiary rounded-3"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-book fa-6x "
      data-icon="book"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M96 0C43 0 0 43 0 96V416c0 53 43 96 96 96H384h32c17.7 0 32-14.3 32-32s-14.3-32-32-32V384c17.7 0 32-14.3 32-32V32c0-17.7-14.3-32-32-32H384 96zm0 384H352v64H96c-17.7 0-32-14.3-32-32s14.3-32 32-32zm32-240c0-8.8 7.2-16 16-16H336c8.8 0 16 7.2 16 16s-7.2 16-16 16H144c-8.8 0-16-7.2-16-16zm16 48H336c8.8 0 16 7.2 16 16s-7.2 16-16 16H144c-8.8 0-16-7.2-16-16s7.2-16 16-16z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <br />
    <h1
      className="text-body-emphasis"
    >
      Dictionary
    </h1>
    <p
      className="col-lg-8 mx-auto fs-5 text-muted"
    >
      This dictionary application allows you to enter a word and it will show you the definition and the example usage. Go ahead and try it!
    </p>
    <div
      className="bd-example"
    >
      <form>
        <div
          className="w-50 p-3"
          style={
            Object {
              "marginLeft": "30%",
            }
          }
        >
          <input
            aria-describedby="emailHelp"
            className="form-control d-inline-flex align-items-center"
            id="userInputWord"
            placeholder="Type a word here to get the definition"
            type="text"
          />
        </div>
      </form>
      <br />
      <div
        className="d-inline-flex gap-2 mb-5"
      >
        <button
          className="d-inline-flex align-items-center btn btn-primary btn-lg px-4 rounded-pill"
          onClick={[Function]}
          type="button"
        >
          Get definition
        </button>
      </div>
      <h4
        className="text-uppercase"
      />
      <span>
        <div>
          Enter a word above and click "Get Definition"
        </div>
      </span>
    </div>
  </div>
</div>
`;
